src/components/explorer/SearchAllRecords.js

```diff
@@ -38,10 +38,13 @@ SearchAllRecords.propTypes = {
     datasetID: PropTypes.string,
     performFreeTextSearchIfPossible: PropTypes.func,
     searchAllRecords: PropTypes.object,
+    isFetchingAdvancedSearch: PropTypes.bool,
+    isFetchingTextSearch: PropTypes.bool
 };
 
-const mapStateToProps = (state) => ({
-    searchAllRecords: state.explorer,
+const mapStateToProps = (state, ownProps) => ({
+    isFetchingAdvancedSearch: state.explorer.fetchingSearchByDatasetID[ownProps.datasetID] || false,
+    isFetchingTextSearch: state.explorer.fetchingTextSearch || false,
```

Comment:
what is the `|| false` for here? 
also for defaulting `undefined` to `false` for the line above, I think `??` is a better operator to use

Reply:
This style is used throughout (see, eg, ExplorerDatasetSearch) so I'm probably guilty of following it without much thought. Switched to `??`

Reply:
yeah I think probably it's a pattern from pre-mainlining of `??` which happened in ~2020